sdk/core/azure-core/test/test_context.cpp

```diff
@@ -0,0 +1,53 @@
+
+
+#include "context.hpp"
+
+using namespace Azure::Core;
+
+struct SomethingUnexpected : ValueBase
+{
+  int a, b, c;
+  std::string d;
+};
+
+
+int main()
+{
+  using namespace std::chrono;
+  using namespace std::chrono_literals;
+  Azure::Core::Context ctx;
+  auto now = system_clock::now();
+  auto ten_minutes_from_now = now + 10min;
+  auto ten_seconds_from_now = now + 10s;
+  auto in_ten_minutes = ctx.WithDeadline(ten_minutes_from_now);
+  auto in_ten_seconds = in_ten_minutes.WithDeadline(ten_seconds_from_now);
+  assert(in_ten_minutes.CancelWhen() == ten_minutes_from_now);
```

Comment:
I think there's already test framework, integrated with pipelines, try using it.